torch/csrc/autograd/engine.cpp

```diff
@@ -454,8 +454,8 @@ void GraphTask::mark_as_completed_and_run_post_processing() {
   if (future_completed_.exchange(true)) {
     // Future is already marked complete, or being marked as such.
     // In case the marking complete is only in progress, we add a
-    // waitNoThrow() to guarantee the future is marked complete on exit.
-    future_result_->waitNoThrow();
+    // wait() to guarantee the future is marked complete on exit.
+    future_result_->wait();
```

Comment:
Is this one never throwing? Shall we handle that?

Reply:
This one doesn't throw, since the ivalue::Future API is slightly different. `wait()` doesn't return the value and it is just void and `value()` actually returns the value and throws an exception.

Reply:
Sounds good!